FastSim Tracking: refactor TrajectorySeedProducer (introduce SeedFinder), clean-up tracking code, fix electron HLT

lveldere opened the pull request:
no changes expected, except for the electron HLT

Commit:
cleanup of fastsim TrajectorySeedProducer

Commit:
got it to run

matt-komm committed:
moved seeding into new class

matt-komm committed:
Merge branch 'fastsim-clean-trajectoryseedprod' of https://github.com/lveldere/cmssw into ext_seeding

Conflicts:
	FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc
	FastSimulation/Tracking/plugins/TrajectorySeedProducer.h

lveldere committed:
Update TrajectorySeedProducer.h

lveldere committed:
Update TrajectorySeedProducer.cc

lveldere committed:
Update TrajectorySeedProducer.h

Commit:
Merged patch-1 from repository lveldere

kovitang committed:
762

kovitang committed:
762

kovitang committed:
working 2D performance for DQM

kovitang committed:
working 2D performance for DQM

pvmulder committed:
cleanup of RECO sequence for b-tagging

pvmulder committed:
cleanup of RECO sequence for b-tagging

Commit:
Add c-taggers to the DQM and input variables. Adapt TagCorrelation to match needs of 2D performance for c-tagger. Some cleaning. Update of the examples scripts.

Commit:
Add c-taggers to the DQM and input variables. Adapt TagCorrelation to match needs of 2D performance for c-tagger. Some cleaning. Update of the examples scripts.

Commit:
Add SISVHE to the DQM.

Commit:
Add SISVHE to the DQM.

Commit:
Add SISVHE cfi file.

Commit:
Add SISVHE cfi file.

Commit:
Small fix for the c-tagger tag info in DQM.

Commit:
Small fix for the c-tagger tag info in DQM.

Commit:
Update c-tagger input variables and 0.5 WP.

Commit:
Update c-tagger input variables and 0.5 WP.

jfernan2 committed:
DTDQM-reading-TwinMux. dttfDigis still reading DCC in Configuration/StandardSequences/python/RawToDigi_cff.py as process dttfDigis, change in future to dttm7Digis

pvmulder committed:
real clean up of reco sequence and RecoBTag/SecondaryVertex

pvmulder committed:
real clean up of reco sequence and RecoBTag/SecondaryVertex

pvmulder committed:
clean up of RecoBTag/Combined

pvmulder committed:
clean up of RecoBTag/Combined

pvmulder committed:
updates

pvmulder committed:
updates

pvmulder committed:
fixes

pvmulder committed:
fixes

Commit:
more cleanup

Commit:
Introduce changes needed for Hcal geometry for 2016 and 2017

Commit:
Update a files

Commit:
update the scenarios as well

Commit:
Do some bug fixing

Commit:
Avoid unit test failure

jfernan2 committed:
Fixed labels on DQM DT evolution plots vs LS using  DTTimeEvolutionHisto class

Commit:
Bug fixes

Commit:
Bug fixing done

Commit:
Take care of Yana's comments

Commit:
Remove confusion

Commit:
Fix maximum depth for SLHC option

Commit:
Extra checks

Commit:
Remove some commented lines + update of the destructor of EffPurFromHistos(2D) classes.

Commit:
Remove some commented lines + update of the destructor of EffPurFromHistos(2D) classes.

Dr15Jones committed:
Configuration parameter copy and modify functions extended

The `clone` and `Modifier.toModify` methods have been extended to allow further modifications to the parameters after they have been copied.
1) Specify `None` as the value of the parameter will cause the parameter to be removed from the configuration
2) Parameters within embedded PSets can now be modified using a python dict as the value. The keys of the dictionary are the parameter
    names and the values are the new values to use.
The `clone` and `Modifier.toModify` both call the same helper function to do the work of modifying the copied parameters.

Commit:
Small example update.

Commit:
Small example update.

wmtan committed:
Sort module summary reports by module label

jfernan2 committed:
Only python cfi changes to be compliant with PR #13165

Commit:
Move to std::unique_ptr

Commit:
Move to std::unique_ptr

Commit:
Merge branch 'bTagRecoAndDQMcleanup' of github.com:cms-btv-pog/cmssw into bTagRecoAndDQMcleanup

Commit:
first merge attempt

Commit:
first running version after merge

Commit:
fix ele HLT

Commit:
debugged

Commit:
Merged refs/pull/13214/head from repository cms-sw

Commit:
merge

Dr15Jones committed:
Fix Modifier to allow adding new parameters

The reimplementation of the Modifier accidentally lost the ability to specify a new parameter when doing the modification. This adds that feature back.

Commit:
debug after merge

Commit:
Merge remote-tracking branch 'my-cmssw/fastsim-trackingcuFeb9' into CMSSW_8_0_X

cmsbuild commented:
A new Pull Request was created by  @lveldere  for CMSSW_8_0_X.

It involves the following packages:

FastSimulation/Tracking


@ssekmen, @lveldere, @civanch, @mdhildreth, @cmsbuild, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @matt-komm this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

lveldere commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

lveldere commented:
@ANSH0712 

Note 
- hit-skipping in the TrackCandidateProducer has been re-written a bit
- TrajectorySeedProducer has changed completely, we probably need to discuss how to do the hit triplet check in this framework. I actually think it will be easier.

cmsbuild commented:
-1 
Tested at: dc20f57083127a19e11d21cd62f581779f446ee4
 I found an error when building: 
 
 <pre>>> Building LCG reflex dict from header file src/DataFormats/L1GlobalTrigger/src/classes.h
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-09-1100/src/DataFormats/L1GlobalTrigger/src/L1GlobalTriggerObjectMap.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-09-1100/src/DataFormats/L1GlobalTrigger/src/L1GlobalTriggerObjectMapRecord.cc 
In file included from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-09-1100/src/DataFormats/L1GlobalTrigger/interface/L1GlobalTriggerEvmReadoutRecord.h:19:0,
                 from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-09-1100/src/DataFormats/L1GlobalTrigger/src/L1GlobalTriggerEvmReadoutRecord.cc:16:
/afs/cern.ch/cms/sw/ReleaseCandidates/volB/slc6_amd64_gcc493/external/gcc/4.9.3/include/c++/4.9.3/string:38:28: error: /afs/cern.ch/cms/sw/ReleaseCandidates/volB/slc6_amd64_gcc493/external/gcc/4.9.3/include/c++/4.9.3/x86_64-redhat-linux-gnu/bits/c++config.h: Bad address
 #include <bits/c++config.h>
                            ^
In file included from /afs/cern.ch/cms/sw/ReleaseCandidates/volB/slc6_amd64_gcc493/external/gcc/4.9.3/include/c++/4.9.3/bits/stringfwd.h:40:0,
                 from /afs/cern.ch/cms/sw/ReleaseCandidates/volB/slc6_amd64_gcc493/external/gcc/4.9.3/include/c++/4.9.3/string:39,
                 from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-09-1100/src/DataFormats/L1GlobalTrigger/interface/L1GlobalTriggerEvmReadoutRecord.h:19,
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13246/11135/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/dc20f57083127a19e11d21cd62f581779f446ee4
https://github.com/cms-sw/cmssw/commit/4b6956e54bc76dc66dda1d7939d2151b53898298
https://github.com/cms-sw/cmssw/commit/c28ed099a407b38c4e5143837e5ad362a2d192d9
https://github.com/cms-sw/cmssw/commit/e5f6a7243e2b6d76d3ba62063491c8227f0a06dc
https://github.com/cms-sw/cmssw/commit/edb1732ccec2abc18297657896f003b757c3bb37
https://github.com/cms-sw/cmssw/commit/b4d612471e8341ff43fc10dee7536b3c1dae4586
https://github.com/cms-sw/cmssw/commit/dec210faa1415fe9949c6c1d8cb0631f677a7c56
https://github.com/cms-sw/cmssw/commit/619e0533c283b93e9ec5a67ac91fc69e4f2cf253
https://github.com/cms-sw/cmssw/commit/cce9f46168f3b214facdc240fc9e48c0593cfdf6
https://github.com/cms-sw/cmssw/commit/751056ed78c2ff8dae834067746fcc820cdea1b8
https://github.com/cms-sw/cmssw/commit/6adfec2e56eb7639c241797f9f8afcb0fc8dd8e5
https://github.com/cms-sw/cmssw/commit/0dc13c70008b8555782c3cee66b06d47fbba95e3
https://github.com/cms-sw/cmssw/commit/3a1a160b852324db55d6b7ac9c75bc62c6f1ff97
https://github.com/cms-sw/cmssw/commit/68a8254ea38022de291eede673ac4696c225fa90
https://github.com/cms-sw/cmssw/commit/459889e5640c27d2956593c0fc5dbda0991e8898
https://github.com/cms-sw/cmssw/commit/36c3ddd7d1508908887f25d49605636676d48bc3
https://github.com/cms-sw/cmssw/commit/51ee9d373e34fde3aa5e249bdb47f4903e2c8dbf
https://github.com/cms-sw/cmssw/commit/4a9dde799d3a8195302a08469b5b37ffe4b408ca
https://github.com/cms-sw/cmssw/commit/d4bf8ef95d5c4ebe55a804b94a11138e0218eac0
https://github.com/cms-sw/cmssw/commit/638172b9d39229de0db1efe3dbe2e008d6d7239a
https://github.com/cms-sw/cmssw/commit/b55fbd842747ae3563d0aeeb6e66651e60d48f25
https://github.com/cms-sw/cmssw/commit/7c1d40335add64a7cd301dc092f48155c67f5c56
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13246/11135/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13246/11135/git-merge-result

cmsbuild committed:
Merge pull request #13225 from wmtan/SortModuleSummaryReports

Sort module summary reports by module label

davidlange6 committed:
Merge pull request #13139 from bsunanda/Run2-hcx64

bsunanda:Run2-hcx64 Introduce changes needed for Hcal geometry for 2016 and 2017

lveldere commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: dc20f57083127a19e11d21cd62f581779f446ee4
 I found an error when building: 
 
 <pre>>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-10-2300/src/FastSimulation/Tracking/src/TrackingLayer.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-10-2300/src/FastSimulation/Tracking/src/TrajectorySeedHitCandidate.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-10-2300/src/FastSimulation/Tracking/test/testGeneralTracks.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-10-2300/src/FastSimulation/Tracking/plugins/ConversionTrackRefFix.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-10-2300/src/FastSimulation/EgammaElectronAlgos/src/FastPixelHitMatcher.cc: In member function 'std::vector<std::pair<std::shared_ptr<const TrackingRecHit>, std::shared_ptr<const TrackingRecHit> > > FastPixelHitMatcher::compatibleHits(const GlobalPoint&, const GlobalPoint&, float, std::vector<TrajectorySeedHitCandidate>&)':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-10-2300/src/FastSimulation/EgammaElectronAlgos/src/FastPixelHitMatcher.cc:135:58: error: '__gnu_cxx::__alloc_traits<std::allocator<TrajectorySeedHitCandidate> >::value_type' has no member named 'geomDet'
  GenericTransientTrackingRecHit::build(theHits[firstHit].geomDet(),
                                                          ^
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-10-2300/src/FastSimulation/EgammaElectronAlgos/src/FastPixelHitMatcher.cc:138:59: error: '__gnu_cxx::__alloc_traits<std::allocator<TrajectorySeedHitCandidate> >::value_type' has no member named 'geomDet'
  GenericTransientTrackingRecHit::build(theHits[secondHit].geomDet(),
                                                           ^
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13246/11139/summary.html

davidlange6 committed:
Merge pull request #13244 from Dr15Jones/extendedPSetModification2

Configuration parameter copy and modify functions extended

Commit:
Implement 'scan once' mode in online DQM input sources.

Completely ignores file "streaming" and uses only files found
at the initial scan. Useful for testing and development.

Commit:
Several fixes for the upload script

Commit:
Add the Isotrack filter for a new stream

Commit:
Add the Isotrack filter for a new stream

Commit:
Also modify autoAlca

Commit:
Remove upliate

Commit:
get better indentation

Commit:
missed modifications

cgalloni committed:
Update PFRecoTauChargedHadronFromTrackPlugin.cc

Change LogWarning to LogInfo in case area subjet is  0

davidlange6 committed:
Merge pull request #13250 from dmitrijus/scan_80x

Implement 'scan once' mode in online DQM input sources

Commit:
remove the ff file errors

ggovi committed:
Payload Inspector target db set to prod. Added SiStrip plots. Cleanup v1 stuff.

Commit:
Follow Marco's suggestions

fcouderc committed:
add non triggering electron Loose WP

cmsbuild committed:
Merge pull request #13259 from ggovi/condition-uploader-script-fix-0

Several fixes for the upload script

Commit:
Take David's comments

davidlange6 committed:
Merge pull request #13261 from ggovi/PayloadInspector-v2

Payload Inspector target DB set to prod

davidlange6 committed:
Merge pull request #13265 from fcouderc/looseMVAWP_NonTriggeringElectrons_80X

add non triggering electron Loose WP

davidlange6 committed:
Merge pull request #13098 from cms-btv-pog/bTagRecoAndDQMcleanup

Clean up of btag sequence in standard reco + update of DQM

davidlange6 committed:
Merge pull request #13123 from jfernan2/DTDQM-reading-TwinMux

DTDQM reading TwinMux.

perrotta committed:
Merged HLTbeyond800pre6 from repository cms-tsg-storm

perrotta committed:
Tracking updates for 2016 HLT

Martin-Grunewald committed:
Merged refs/pull/13270/head from repository cms-sw

Martin-Grunewald committed:
Cleanup of 2016 tracking custom files

cmsbuild committed:
Merge pull request #13269 from cgalloni/80ReMiniAODwBoostedTaus_forPull_v800pre4

Update PFRecoTauChargedHadronFromTrackPlugin.cc

cmsbuild committed:
Merge pull request #13270 from cms-tsg-storm/HLTbeyond800pre6

Tracking updates for 2016 HLT

fwyzard committed:
remove unused variables

davidlange6 committed:
Merge pull request #13251 from bsunanda/Run2-alca35

bsunanda:Run2 alca35 Add IsoTrackFilter for normal physics events

cmsbuild committed:
Merge pull request #13276 from fwyzard/HLTRPCTrigNoSyncFilter_remove_unused_variables

HLTRPCTrigNoSyncFilter: remove unused variables

lveldere commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: dc20f57083127a19e11d21cd62f581779f446ee4
 I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-15-1100/src/FastSimulation/EgammaElectronAlgos/src/FastPixelHitMatcher.cc 
Copying tmp/slc6_amd64_gcc493/src/FastSimulation/Tracking/src/FastSimulationTracking/libFastSimulationTracking.so to productstore area:
Leaving library rule at FastSimulation/Tracking
>> Building edm plugin tmp/slc6_amd64_gcc493/src/FastSimulation/Tracking/test/testGeneralTracks/libtestGeneralTracks.so
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-15-1100/src/FastSimulation/Muons/plugins/FastTSGFromIOHit.cc:17:0:
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-15-1100/poison/FastSimulation/Tracking/interface/FastTrackingHelper.h:1:2: error: #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
 #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
  ^
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-15-1100/src/FastSimulation/Muons/plugins/FastTSGFromPropagation.cc:43:0:
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-15-1100/poison/FastSimulation/Tracking/interface/FastTrackingHelper.h:1:2: error: #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
 #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13246/11222/summary.html

Commit:
Merged refs/pull/13246/head from repository cms-sw

Commit:
get FastSimulation/Muon to compile

Commit:
remove obsolete package FastSimulation/EgammaElectronAlgos (it no longer compiles)

lveldere commented:
needs rebase

cmsbuild commented:
Pull request #13246 was updated. @diguida, @lveldere, @ianna, @cerminar, @Martin-Grunewald, @vanbesien, @perrotta, @civanch, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @deguio, @slava77, @ggovi, @vadler, @mmusich, @ssekmen, @monttj, @franzoni can you please check and sign again.
